common/environment/environment.cpp

```diff
@@ -31,11 +31,77 @@
 #include "dalienv.hpp"
 
 #define SDS_LOCK_TIMEOUT  30000
-
+#define DEFAULT_DROPZONE_INDEX      1
+#define DROPZONE_BY_MACHINE_SUFFIX  "-dropzoneByMachine-"
+#define DROPZONE_SUFFIX             "dropzone-"
+#define MACHINE_PREFIX              "machine-"
+#define NUMBER_OF_DROPZONES_SUFFIX  "-numberOfDropZones"
 
 static int environmentTraceLevel = 1;
 static Owned <IConstEnvironment> cache;
 
+class CLocalEnvironment;
+
+class CConstMachineInfoIterator : public CInterface, implements IConstMachineInfoIterator
+{
+public:
+    CConstMachineInfoIterator();
+    IMPLEMENT_IINTERFACE
+    bool first();
+    bool next();
+    bool isValid();
+    virtual IConstMachineInfo & query();
+    virtual unsigned count();
+
+protected:
+    IConstMachineInfo * curr;
+    Owned<CLocalEnvironment> constEnv;
+    unsigned index;
+    unsigned maxIndex;
+};
+
+class CConstDropZoneInfoIterator : public CInterface, implements IConstDropZoneInfoIterator
+{
+public:
+    CConstDropZoneInfoIterator();
+    IMPLEMENT_IINTERFACE
+    bool first();
+    bool next();
+    bool isValid();
+    virtual IConstDropZoneInfo & query();
+    virtual unsigned count();
+    virtual IConstDropZoneInfo * ptr();
```

Comment:
See comment in definition of IConstDropZoneInfoIterator - why does this iterator need a ptr() method, when query() is the standard way to ask for item of iterator?